src/components/toc.js

```diff
@@ -26,39 +23,41 @@ const Toc = ({ toc }) => {
 const createIntersectionObserver = () => {
 	// gripes with intersection observer: https://twitter.com/nipafx/status/1217590444772810753
 
-	// to make sure only events after the initial observation are acted on,
-	// store the ids of visited headers
-	const visited = []
-	const update = events =>
-		events.forEach(event => {
-			const id = event.target.querySelector(`span`).id
-			if (visited.includes(id)) {
-				// `event.isIntersecting` should indicate whether the element
-				// transitioned into or out of intersection, but Firefox instead
-				// returns true if intersection ratio is > 0;
-				// instead fall back to comparing intersection ratio with threshold
-				const visible = event.intersectionRatio < INTERSECTION_THRESHOLD
-				if (visible) {
-					const position = event.boundingClientRect.top
-					updateHighlight(id, position)
-				}
-			} else visited.push(id)
-		})
-	return new IntersectionObserver(update, {
+	// registering the intersection observer triggers an event for each
+	// observed header, but none of them should be highlighted at this point;
+	// instead store the initial events' targets and only act on an event
+	// if its target was stored earlier.
+	const visited = {}
+
+	const updateHighlights = event => {
+		const focusedHeaderId = event.target.querySelector(`span`).id
```

Comment:
I changed `entry` back to `event` because these are intersection observer events, not the observed entries themselves. (This becomes clear, when calling `event.target` - `entry.target` doesn't make much sense.)